src/helpers/await.ts

```diff
@@ -24,34 +24,47 @@ export type AsyncFunction<T> = () => Awaitable<AsyncData<T>>;
 export function getAbortSignal(
   signal?: AbortSignal,
   maxTimeout: number = 60_000
-): AbortSignal {
+): {
+  signal: AbortSignal;
+  timeoutId: ReturnType<typeof setTimeout> | undefined;
+} {
   let abortSignal: AbortSignal;
+  let timeoutId;
   if (signal == null) {
     const controller = new AbortController();
     abortSignal = controller.signal;
-    setTimeout(controller.abort.bind(controller), maxTimeout);
+    // return the timeoutId so the caller can cleanup
+    timeoutId = setTimeout(function () {
+      controller.abort();
```

Comment:
I still can't say for sure what is causing the tests to occasionally fail, but this change seems good in general.  If the request has succeeded, or the controller has already been aborted, we don't need to be waiting for this timeout and eventually calling abort.